src/IECore/VectorTypedData.cpp

```diff
@@ -289,16 +289,17 @@ void BoolVectorData::load( LoadContextPtr context )
 template<>
 void ShortVectorData::save( Object::SaveContext *context ) const
 {
+	const std::vector<short> &s = readable();
 	Data::save( context );
 	IndexedIOPtr container = context->container( staticTypeName(), 1 );
-	container->write( g_valueEntry, baseReadable(), baseSize() );
+	container->write( g_valueEntry, &(s[0]), s.size() );
```

Comment:
> I'm not clear why this is necessary. That should be the same as calling baseReadable() and baseSize()

It seems we're not calling `IE_CORE_DEFINEVECTORTYPEDDATATRAITSSPECIALIZATION( ShortData )` anywhere, meaning `baseReadable()` and `baseWritable()` aren't defined properly. We should fix that and remove these changes.

> There also seems to be a note on line 206 saying this was intentionally saving/loading ints... I'm not sure why that would have been though. Any recollection @johnhaddon ?

A bit of git archeology implies that it was simply because IndexedIO didn't have native support for shorts at that time. We changed to storing the native data in 9adee91be390e91ce8cd5faa6ec9e36ca9e5f91e, back in 2008.

> If the rawContainer does work, I wonder if we even need this specialization anymore or if we can call the macro like the other simple types.

The `container()/rawContainer()` distinction is separate to this bug, with the latter being an optimisation that creates fewer IndexedIO directories at the expense of some namespacing. I believe the default specialisation will work, but only if you're happy to drop compatibility for pre-2008 files.

Reply:
@andrewkaufman was fine breaking compatibility here, so I just removed the whole specialisation.